tests/test_core.py

```diff
@@ -145,21 +145,17 @@ def attempt_terminate():
 
     def test_irregular_tablenames(self):
         """Irregular table names need to be quoted"""
-        db = SqliteDict(':memory:', tablename='9nine')
-        db['key'] = 'value'
-        db.commit()
-        self.assertEqual(db['key'], 'value')
-        db.close()
+        def __test_irregular_tablenames(tablename):
+            db = SqliteDict(':memory:', tablename=tablename)
+            db['key'] = 'value'
+            db.commit()
+            self.assertEqual(db['key'], 'value')
+            db.close()
 
-        db = SqliteDict(':memory:', tablename='outer space')
-        db['key'] = 'value'
-        db.commit()
-        self.assertEqual(db['key'], 'value')
-        db.close()
+        __test_irregular_tablenames('9nine')
+        __test_irregular_tablenames('outer space')
+        __test_irregular_tablenames('table with a "quoted" name')
```

Comment:
How about some non-ASCII characters?

Reply:
I don't know whether names are strings (as in unicode) or bytes. I'm assuming in 2.x it's bytes and 3.x it's unicode

I noticed there were a few bug requests mentioning problems using unicode strings as keys.

I added a test for that.

Reply:
Here they are clearly strings. Do you mean inside sqlite, internally?

Reply:
I mean that in python3 calling with tablename=b'something' fails, it has to be a string, wheres in python2 calling with tablename=u'something' _might_ fail.